submissions/scripts/prep_files.py

```diff
@@ -0,0 +1,460 @@
+"""
+Prepares files and directories for sequencing submissions.
+
+"""
+import os
+import re
+import sys
+import argparse
+import tarfile
+from pathlib import Path
+from datetime import datetime
+import logging
+import pandas as pd
+import conda.cli.python_api as Conda
+import docker
+from openpyxl import load_workbook
+from extract_sfs_identifiers import read_all_identifiers, find_sfs_identifiers
+import requests
+import getpass
+
+from export_lims_metadata import add_lims_metadata
+
+LOG_LEVEL = os.environ.get("LOG_LEVEL", "debug").upper()
+
+logging.basicConfig(
+    level = logging.ERROR,
+    format = "[%(asctime)s] %(levelname)-8s %(message)s",
+    datefmt = "%Y-%m-%d %H:%M:%S%z")
+
+logging.captureWarnings(True)
+
+LOG = logging.getLogger(__name__)
+LOG.setLevel(LOG_LEVEL)
+
+def yes_no_cancel(message):
+    while True:
+        user_input = input(f"{message} [y]es/[n]o/[c]ancel: ").lower()
+        if user_input not in ['y','n','c']:
+            print("Not a valid response")
+            continue
+        else:
+            break
+    if user_input == 'y':
+        return True
+    elif user_input == 'n':
+        return False
+    else:
+        print("Cancelling.")
+        sys.exit()
+
+def count_s_occurrences(values):
+    count = 0
+    if not isinstance(values, list):
+        return count
+    for item in values:
+        if isinstance(item, str) and item.lower().strip().startswith("s:"):
+            count += 1
+    return count
+
+def standardize_date(date_value):
+    if pd.isna(date_value) or (isinstance(date_value, str) and date_value.lower().strip() == 'na'):
+        return None
+    else:
+        # So far, two date formats have been encountered in metadata Excel files, resulting in different formats in
+        # resulting dataframes. If `date_value` does not match either of these formats, a ValueError exception will be
+        # thrown. If the rejected date is in a usable format, it should be added here.
+        datetime_val = pd.to_datetime(date_value, format='%Y-%m-%d %H:%M:%S', errors="coerce")
+        if pd.isnull(datetime_val):
+            datetime_val = pd.to_datetime(date_value, format='%m/%d/%Y')
+
+        # The minus sign before m and d removes leading zeros, to match the format of the original Excel file.
+        # Note: this may only work on unix, windows may need to use `#` instead of `-`
+        return datetime_val.strftime('%-m/%-d/%Y')
+
+
+def standardize_barcode(barcode):
+    if pd.isna(barcode):
+        return None
+    elif isinstance(barcode, str):
+        standardized_barcode = barcode.strip().lower()
+    elif isinstance(barcode, int):
```

Comment:
Haven't seen that happen yet, but I'll keep an eye out for it. When that does occur, can it be cast to str or does it need reformatting?

Reply:
I want to say you have to be extra careful with them because its primarily barcodes with the format `123456e9` that convert to exponents. With that said, I haven't seen it happen with new new Excel reader. Mostly just a warning I guess, this code will catch it if it comes up.